### ISSUE TITLE
Spyder claims to be Connecting to kernel while running a file

### ISSUE THREAD

**Issue by impact27** [spyder-ide/spyder#13048]
### Issue Report Checklist

* [ ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ ] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [ ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Spyder claims to be Connecting to kernel while running a file

### What steps reproduce the problem?

1.  create a file with:
```
import time
time.sleep(10)
print('Done')
```
2. Open a new kernel
3. Before the kernel is loaded, press the "Run File" button

### What is the expected output? What do you see instead?
I expect spyder to connect, then run the file. Instead, "Connecting to kernel" is shown until the processing is finished.

## Versions

* Spyder version: 4.x
* Python version: 3.7.7
* Qt version: Qt 5.12.6
* PyQt version: PyQt5 5.12.3
* Operating System name/version: Darwin 19.5.0 

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by bcolsen** [comment]
I can confirm this happens(I had to use the run hot key because my kernel started too soon)

This doesn't seem to happen on kernel restarts. Nothing is run.

> I expect spyder to connect, then run the file. Instead, "Connecting to kernel" is shown until the processing is finished.

The command cache would help here. The console can check for cached commands when it's ready. 

How does the "run in dedicated console" deal with this? It opens and console and runs right away.

**Comment by dalthviz** [comment]
Thanks @impact27 for finding the issue and @bcolsen for the confirmation. I was able to reproduce this too :+1:

**Comment by ccordoba12** [comment]
> I expect spyder to connect, then run the file. Instead, "Connecting to kernel" is shown until the processing is finished.

Most probably an execute message is sent to the kernel before the first prompt is shown, which prevents us from hiding the loading page (because that page is hidden as soon as the first prompt appears).

However, I don't know how to solve this.

**Comment by impact27** [comment]
Maybe the kernel sends the header text when it is ready? (The python version and all) Alternatively we could disable the execute function until the console is ready

**Comment by ccordoba12** [comment]
> Alternatively we could disable the execute function until the console is ready

Yeah, I think that's a simpler and easier way to fix this.

**Comment by ccordoba12** [comment]
Thanks a lot @impact27 for your help with this one! I just released qtconsole 4.7.5 with your fix there.
